### ISSUE TITLE
Validation of node.environment fails if you set he value via an environment variable like ${ENV:xxx}

### ISSUE THREAD

**Issue by Kyle-Payne** [trinodb/trino#9016]
The node.environment only allows lower case letters, numbers and an underscore. The current validation checks don't account for the fact that you may want to set the node.environment via an environment variable so the following results in a malformed text error: node.environment=${ENV:SEP_ENV}.
Another engineer suggested an easy fix is to allow ${ENV:[a-z]} in the regex that validates values of node.environment.

**Comment by findepi** [comment]
I guess you're setting `node.environment` in `node.properties`?
if i read the code correctly, this is read with `launcher.py` https://github.com/airlift/airlift/blob/b29ae98ef532a963bf1475090c7698998582de10/launcher/src/main/scripts/bin/launcher.py#L436
then passed to JVM https://github.com/airlift/airlift/blob/b29ae98ef532a963bf1475090c7698998582de10/launcher/src/main/scripts/bin/launcher.py#L223
and so "overrides" values from config file, and the ENV placeholders are not getting expanded.

(that's my guess, needs further checking)

@electrum might know better

**Comment by nineinchnick** [comment]
If env expansion is done here https://github.com/trinodb/trino/blob/bda8eb55ed49406ef2503e13b00e73ecc54d1dff/core/trino-main/src/main/java/io/trino/server/Server.java#L116 but the node module gets instantiated a few lines above, the validation happens there. Would expanding the validation regex help and would it still be expanded correctly?

**Comment by findepi** [comment]
> Would expanding the validation regex help and would it still be expanded correctly?

That looks like a wrong direction. The validation should run -- and i hope it _is_ run -- on actual bound values, regardless of how they are bound.
Thus, if the validation complains about `${ENV:xxx}`, i would expect `${ENV:xxx}` to be the actual value bound, not some intermediate artifact.

**Comment by nineinchnick** [comment]
You're right, the final values should be validated. If the expansion would need to be moved out of the Bootstrap class, then this issue would be related to https://github.com/trinodb/trino/issues/8245.

**Comment by findepi** [comment]
> then this issue would be related to #8245.

that issue, about `event-listener.properties`, will likely have different solution. Agree that's related still.
